tests/unit/phpDocumentor/Descriptor/Builder/Reflector/FunctionAssemblerTest.php

```diff
@@ -13,50 +13,39 @@
 
 namespace phpDocumentor\Descriptor\Builder\Reflector;
 
-use InvalidArgumentException;
-use Mockery as m;
-use Mockery\Adapter\Phpunit\MockeryTestCase;
 use phpDocumentor\Descriptor\ArgumentDescriptor;
-use phpDocumentor\Descriptor\PackageDescriptor;
 use phpDocumentor\Descriptor\ProjectDescriptorBuilder;
 use phpDocumentor\Reflection\DocBlock;
 use phpDocumentor\Reflection\Fqsen;
 use phpDocumentor\Reflection\Php\Argument;
 use phpDocumentor\Reflection\Php\Function_;
-use function get_class;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument as ProphecyArgument;
+use Prophecy\Prophecy\ObjectProphecy;
 
-class FunctionAssemblerTest extends MockeryTestCase
+class FunctionAssemblerTest extends TestCase
 {
     /** @var FunctionAssembler $fixture */
     protected $fixture;
 
-    /** @var ArgumentAssembler|m\MockInterface */
+    /** @var ArgumentAssembler|ObjectProphecy */
     protected $argumentAssemblerMock;
 
-    /** @var ProjectDescriptorBuilder|m\MockInterface */
+    /** @var ProjectDescriptorBuilder|ObjectProphecy */
     protected $builderMock;
 
     /**
      * Creates a new fixture to test with.
      */
     protected function setUp() : void
     {
-        $this->builderMock = m::mock(ProjectDescriptorBuilder::class);
-        $this->builderMock->shouldReceive('buildDescriptor')->andReturnUsing(
-            static function ($value) {
-                switch (get_class($value)) {
-                    case DocBlock\Tags\Generic::class && $value->getName() === 'package':
-                        return new PackageDescriptor();
-                    default:
-                        throw new InvalidArgumentException('didn\'t expect ' . get_class($value));
-                }
-            }
-        );
-        $this->argumentAssemblerMock = m::mock(ArgumentAssembler::class);
-        $this->argumentAssemblerMock->shouldReceive('getBuilder')->andReturnNull();
-        $this->argumentAssemblerMock->shouldReceive('setBuilder')->with($this->builderMock);
-        $this->fixture = new FunctionAssembler($this->argumentAssemblerMock);
-        $this->fixture->setBuilder($this->builderMock);
+        $this->builderMock = $this->prophesize(ProjectDescriptorBuilder::class);
+        $this->builderMock->buildDescriptor(ProphecyArgument::any(), ProphecyArgument::any())->shouldBeCalled();
```

Comment:
This mock has been changed in behavior... however no tests are failing I think it would be good to keep it as it was. Because right now there is a validation missing, which was here because of some failures. 
The expected argument should be of type `DocBlock\Tags\Generic::class` and has a name `package` 

You could do this with `Argument::that(callback)`

Reply:
Done.